internal/loadbalancer/manager.go

```diff
@@ -27,25 +28,29 @@ type Manager interface {
 }
 
 // ManagerForCluster returns the proper Manager based on the lxcCluster spec.
-func ManagerForCluster(cluster *clusterv1.Cluster, lxcCluster *infrav1.LXCCluster, lxcClient *lxc.Client) Manager {
+func ManagerForCluster(cluster *clusterv1.Cluster, lxcCluster *infrav1.LXCCluster, lxcClient *lxc.Client, managementClusterClient client.Client) Manager {
 	switch {
 	case lxcCluster.Spec.LoadBalancer.LXC != nil:
 		return &managerLXC{
-			lxcClient:        lxcClient,
-			clusterName:      cluster.Name,
-			clusterNamespace: cluster.Namespace,
+			lxcClient:               lxcClient,
+			managementClusterClient: managementClusterClient,
+			clusterName:             cluster.Name,
+			clusterNamespace:        cluster.Namespace,
```

Comment:
and just add a new field here `customHAproxyConfigTemplate`, which you can retrieve from `lxcCluster.Spec.LoadBalancer.LXC.CustomHAProxyConfigTemplate`

Reply:
yes, let's do a simple `string` (avoid `*string`)